tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/CodeOwnerEntry.cs

```diff
@@ -12,81 +14,105 @@ internal class CodeOwnerEntry
-                return null;
+                return !string.IsNullOrEmpty(PathExpression);
             }
+        }
+        private static IEnumerable<string> SplitLine(string line)
+        {
+            // Split the line into segments that are delimited by '@', '%' and the end of the string.
 
-            List<string> entries = SplitLine(line).ToList();
-
-            if (!entries.Any())
+            int previousSplit = -1;
+            for (int i = 0; i < line.Length; i++)
             {
-                return null;
+                if (line[i] == '@' || line[i] == '%')
+                {
+                    if (previousSplit != -1)
+                    {
+                        yield return line.Substring(previousSplit, i - previousSplit).Trim();
+                    }
+                    previousSplit = i;
+                }
             }
 
-            CodeOwnerEntry coe = new CodeOwnerEntry
-            {
-                // the first entry is the path/regex
-                PathExpression = entries[0],
-                ContainsWildcard = entries[0].Contains('*')
-            };
+            // add the last entry
+            yield return line.Substring(previousSplit, line.Length - previousSplit).Trim();
+        }
 
-            // remove the '/' from the path
-            if (coe.PathExpression.StartsWith("/"))
+        public override string ToString()
+        {
+            return $"RegEx:{ContainsWildcard} Expression:{PathExpression} Owners:{string.Join(',', Owners)}  Labels:{string.Join(',', Labels)}";
+        }
+
+        internal static void ParseLabels(CodeOwnerEntry entry, string line)
+        {
+            // Parse a line that looks like # PRLabel: %Label, %Label
+
+            // If we don't have a ':', nothing to do
+            int colonPosition = line.IndexOf(':');
+            if (colonPosition == -1)
             {
-                coe.PathExpression = coe.PathExpression.Substring(1);
+                return;
             }
 
-            for (int i = 1; i < entries.Count; i++)
+            line = line.Substring(colonPosition + 1).Trim();
+            foreach (string label in SplitLine(line).ToList())
             {
-                string entry = entries[i].Trim();
-
-                if (entry[0] == '@')
+                if (label[0] == '%')
                 {
-                    coe.Owners.Add(entry.Substring(1));
+                    entry.Labels.Add(label.Substring(1));
                 }
 
-                if (entry[0] == '%')
-                {
-                    coe.Labels.Add(entry.Substring(1));
-                }
             }
-
-            return coe;
         }
 
-        private static IEnumerable<string> SplitLine(string line)
+        internal static void ParseOwnersAndPath(CodeOwnerEntry entry, string line)
         {
-            // Split the line into segments that are delimited by '@', '%' and the end of the string.
+            if (string.IsNullOrEmpty(line) || line.StartsWith('#'))
```

Comment:
FYI we also have some codeowners parsing logic in PS form https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/get-codeowners.ps1#L23 but I think that logic can probably be replicated here as well.

Reply:
The line just above is interesting: https://github.com/Azure/azure-sdk-tools/blob/24c2a3eb669fa9a0c21e0b95c28c745c22821b99/eng/common/scripts/get-codeowners.ps1#L21

We should probably change that?

As for the code duplication -- I think the code is rather simple and it is fine if we duplicate it here.

Reply:
The comment was added because I had them fix it based on our old label convention. The comment doesn't hurt anything though and the implementation is still good.

Reply:
I just remembered we also already have https://github.com/Azure/azure-sdk-tools/blob/master/tools/notification-configuration/github-codeowner-subscriber/CodeOwnersParser.cs which we should try to share.

Reply:
Yeah... I looked at that... probably should have used that one... too late now :(.

Reply:
I'm fine proceeding but I would like us to at least file an issue to try and combine so we don't keep adding different implementations of code that parse CODEOWNERS files.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/824